### PR TITLE
Added example to :class:`~.SurroundingRectangle`

### PR DESCRIPTION
**Pull request by MathInvariance**
## Explanation for Changes

[Documentation Reference](https://manimce--1689.org.readthedocs.build/en/1689/reference/manim.mobject.shape_matchers.SurroundingRectangle.html)

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [ ] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

File: manim/mobject/shape_matchers.py [hunk 1]
```diff
@@ -10,6 +10,23 @@
...
+    Examples
+    --------
+
+    .. manim:: SurroundingRectExample
+        :save_last_frame:
+
+        class SurroundingRectExample(Scene):
+            def construct(self):
+                title = Title("A Quote from Newton")
+                quote = Text("If I have seen further than others, \nit is by standing upon the shoulders of giants.", color=BLUE).scale(0.75)
```
Comment (PhotonSpheres) [inline manim/mobject/shape_matchers.py:24]:
```suggestion
                quote = Text("If I have seen further than others, \\nit is by standing upon the shoulders of giants.", color=BLUE).scale(0.75)
```

Comment (PhotonSpheres) [reply manim/mobject/shape_matchers.py:24]: The way the doc-making process works, one backslash is taken away. So adding another should fix it and all checks should pass now.

**Review by PhotonSpheres — CHANGES_REQUESTED** [review]

File: manim/mobject/shape_matchers.py (hunk 1 above)
Comment (MathInvariance) [reply manim/mobject/shape_matchers.py]: Thanks for pointing that out.

**Comment by behackl** [comment]
[Link to added example](https://manimce--1689.org.readthedocs.build/en/1689/reference/manim.mobject.shape_matchers.SurroundingRectangle.html?highlight=surrounding#manim.mobject.shape_matchers.SurroundingRectangle)

File: manim/mobject/shape_matchers.py [hunk 2]
```diff
@@ -10,6 +10,23 @@
 
 
 class SurroundingRectangle(Rectangle):
+    """A rectangle surrounding a `mobject`
```
Comment (behackl) [inline manim/mobject/shape_matchers.py:13]:
```suggestion
    """A rectangle surrounding a :class:`~.Mobject`
```

File: manim/mobject/shape_matchers.py [hunk 3]
```diff
@@ -10,6 +10,23 @@
...
+    Examples
+    --------
+
+    .. manim:: SurroundingRectExample
+        :save_last_frame:
+
+        class SurroundingRectExample(Scene):
+            def construct(self):
+                title = Title("A Quote from Newton")
+                quote = Text("If I have seen further than others, \\nit is by standing upon the shoulders of giants.", color=BLUE).scale(0.75)
```
Comment (behackl) [inline manim/mobject/shape_matchers.py:24]: This line is a bit too long, could you please break it up? Also, the escaped backslash actually confuses me, maybe using an r string would work as well instead?

```suggestion
                quote = Text(
                    r"If I have seen further than others, \n"
                    r"it is by standing upon the shoulders of giants.",
                    color=BLUE
                ).scale(0.75)
```

**Review by behackl — CHANGES_REQUESTED** [review]
Added some minor comments, the example looks good to me overall.

File: manim/mobject/shape_matchers.py [hunk 4]
```diff
@@ -10,6 +10,25 @@
...
+    :save_last_frame:
+
+        class SurroundingRectExample(Scene):
+            def construct(self):
+                title = Title("A Quote from Newton", scale_factor=1.5)
+                line1 = "If I have seen further than others,\n"
+                line2 = "it is by standing upon the shoulders of giants."
+                quote = Text(line1 + line2, color=BLUE, size=0.8)
+                box = SurroundingRectangle(quote, color=YELLOW, buff=MED_SMALL_BUFF)
+                self.add(title, box, quote)
```
Comment (behackl) [inline manim/mobject/shape_matchers.py:28]:
```suggestion
        class SurroundingRectExample(Scene):
            def construct(self):
                title = Title("A Quote from Newton")
                quote = Text(
                    "If I have seen further than others, \n"
                    "it is by standing upon the shoulders of giants.",
                    color=BLUE
                ).scale(0.75)
                box = SurroundingRectangle(quote, color=YELLOW, buff=MED_LARGE_BUFF)
                self.add(title, box, quote)
```

**Review by behackl — CHANGES_REQUESTED** [review]
I've tested the suggested formatting locally. I'll apply the suggestion so that we can see whether the documentation build is also happy.

File: manim/mobject/shape_matchers.py [hunk 5]
```diff
@@ -10,6 +10,27 @@
 
 
 class SurroundingRectangle(Rectangle):
+    """A rectangle surrounding a :class:`~.Mobject`
```
Comment (behackl) [inline manim/mobject/shape_matchers.py:13]:
```suggestion
    r"""A rectangle surrounding a :class:`~.Mobject`
```

File: manim/mobject/shape_matchers.py [hunk 6]
```diff
@@ -10,6 +10,27 @@
 
 
 class SurroundingRectangle(Rectangle):
+    r"""A rectangle surrounding a :class:`~.Mobject`
+
+    Examples
+    --------
+
+    .. manim:: SurroundingRectExample
+    :save_last_frame:
```
Comment (hydrobeam) [inline manim/mobject/shape_matchers.py:19]:
```suggestion
        :save_last_frame:
```

**Review by hydrobeam — APPROVED** [review]
Looks good to me 👏Thank you for the contribution!

**Review by behackl — APPROVED** [review]

**Pull request merged**